tests/publiccloud/ipa.pm

```diff
@@ -36,9 +36,18 @@ sub run {
         my $system_max_boot_time = 120;
         my $out                  = script_output('grep "^Startup finished in" ' . $ipa->{logfile});
         record_info('Startup time', $out);
-        die 'Fail to find boot time in log' unless $out =~ /Startup finished in (\d{1,4}\.\d{3})s \(kernel\) \+ \d{1,4}\.\d{3}s \(initrd\) \+ \d{1,4}\.\d{3}s \(userspace\) = (\d{1,4}\.\d{3})s/;
-        record_info('Kernel boot is too slow',         result => 'fail') if $1 > $kernel_max_boot_time;
-        record_info('Overall system boot is too slow', result => 'fail') if $2 > $system_max_boot_time;
+        my $kernel_real_boot_time, $system_real_boot_time;
+        if ($out =~ /\= \d{1,2}min/) {
+            die 'Fail to find boot time in log' unless $out =~ /Startup finished in (\d{1,4}\.\d{3})s \(kernel\) \+ \d{1,4}\.\d{3}s \(initrd\) \+ \d{1,4}\.\d{3}s \(userspace\) = (\d{1,2})min (\d{1,4}\.\d{3})s/;
+            $kernel_real_boot_time = $1;
+            $system_real_boot_time = $2 * 60 + $3;
+        } else {
+            die 'Fail to find boot time in log' unless $out =~ /Startup finished in (\d{1,4}\.\d{3})s \(kernel\) \+ \d{1,4}\.\d{3}s \(initrd\) \+ \d{1,4}\.\d{3}s \(userspace\) = (\d{1,4}\.\d{3})s/;
```

Comment:
Since you could match once and figure out if $1 or $4 have your suspect, you can just check if that one has the minute and then blame... Or something else could be even more efficient :) but it's a start I guess...